src/git-auth-helper.ts

```diff
@@ -109,14 +114,40 @@ class GitAuthHelper {
     } else {
       await fs.promises.writeFile(newGitConfigPath, '')
     }
-
     try {
       // Override HOME
       core.info(
         `Temporarily overriding HOME='${this.temporaryHomePath}' before making global git config changes`
       )
       this.git.setEnvironmentVariable('HOME', this.temporaryHomePath)
+    } catch (err) {
+      core.info('Failed to clone global config')
+      throw err
+    }
+
+    // Setup the workspace as a safe directory, so if we pass this into a container job with a different user it doesn't fail
+    // Otherwise all git commands we run in a container fail
+    core.info(
+      `Adding working directory to the temporary git global config as a safe directory`
+    )
+    await this.git
+      .config(
```

Comment:
does the `.config` handles the `--add` for you?

Reply:
Yeah, the fourth parameter is add

```
  config(
    configKey: string,
    configValue: string,
    globalConfig?: boolean,
    add?: boolean
```